src/ordered_set_lang.mli

```diff
@@ -62,21 +74,17 @@ module Unexpanded : sig
 
   val has_special_forms : t -> bool
 
-  (** List of files needed to expand this set *)
-  val files
-    : t
-    -> f:(String_with_vars.t -> Path.t)
-    -> Sexp.syntax * Path.Set.t
+  type expander =
+    { f: 'a. mode:'a String_with_vars.Mode.t
+        -> String_with_vars.t
+        -> ('a, String_with_vars.t) String_with_vars.Partial.t
+    }
 
-  (** Expand [t] using with the given file contents. [file_contents] is a map from
-      filenames to their parsed contents. Every [(:include fn)] in [t] is replaced by
-      [Map.find files_contents fn]. Every element is converted to a string using [f]. *)
   val expand
     :  t
     -> dir:Path.t
-    -> files_contents:Sexp.Ast.t Path.Map.t
-    -> f:(String_with_vars.t -> Value.t list)
-    -> expanded
+    -> f:expander
+    -> Partial.t * Sexp.syntax * Path.Set.t
```

Comment:
It feels a bit weird to return the syntax here, what about adding a function `val syntax : t -> Sexp.syntax`?